tests/ImageSharp.Tests/Image/ImageTests.WrapMemory.cs

```diff
@@ -80,6 +80,52 @@ public override void Unpin()
                 }
             }
 
+            public sealed class CastMemoryManager<TFrom, TTo> : MemoryManager<TTo>
+                where TFrom : unmanaged
+                where TTo : unmanaged
+            {
+                private readonly Memory<TFrom> memory;
+
+                public CastMemoryManager(Memory<TFrom> memory)
+                {
+                    this.memory = memory;
+                }
+
+                /// <inheritdoc/>
+                protected override void Dispose(bool disposing)
+                {
+                }
+
+                /// <inheritdoc/>
+                public override Span<TTo> GetSpan()
+                {
+                    if (MemoryMarshal.TryGetArray(this.memory, out ArraySegment<TFrom> arraySegment))
+                    {
+                        return MemoryMarshal.Cast<TFrom, TTo>(arraySegment.AsSpan());
+                    }
+
+                    if (MemoryMarshal.TryGetMemoryManager<TFrom, MemoryManager<TFrom>>(this.memory, out MemoryManager<TFrom> memoryManager))
+                    {
+                        return MemoryMarshal.Cast<TFrom, TTo>(memoryManager.GetSpan());
+                    }
+
+                    ThrowHelper.ThrowArgumentException("The input Memory<byte> instance was not valid.", nameof(this.memory));
+
+                    return default;
+                }
+
+                /// <inheritdoc/>
+                public override MemoryHandle Pin(int elementIndex = 0)
+                {
+                    return this.memory.Pin();
```

Comment:
Same as above

Reply:
Fixed in https://github.com/SixLabors/ImageSharp/pull/1314/commits/eb229964e698852693a8a28748020da93f2c3f22.